src/parser/open-api/allOfResolver.js

```diff
@@ -66,7 +65,7 @@ function resolveAllOfRecursive (obj) {
  * @return {Object} definitions object that has allOf resolved
  */
 export function resolveAllOf (obj) {
-  const clonedObj = clone(obj)
+  const clonedObj = cloneDeep(obj)
```

Comment:
I disagreed with this one. This will slow it down because of deep cloning, for the benefit of peace of mind.

However, if you look at the existing code, wherever there is is a potential change, we do a `cloneDeep` (e.g. line 12). This means we only do cloneDeep where it's needed (i.e. modified). This is similar to the approach used in [immutable](https://www.youtube.com/watch?v=I7IdS-PbEgI&feature=youtu.be), albeit less elegant.